lib/context.ml

```diff
@@ -26,38 +29,40 @@ let init_context line_width (w, h) axes =
 
   let surface = Cairo.Image.create Cairo.Image.ARGB32 ~w ~h in
   let ctx = Cairo.create surface in
-  Cairo.scale ctx (float_of_int w) (float_of_int h);
   Cairo.set_line_width ctx line_width;
+  Cairo.translate ctx (w / 2 |> float_of_int) (h / 2 |> float_of_int);
   context := Some { ctx; surface; size = (w, h); axes }
 
 let resolution () = match !context with Some ctx -> ctx.size | None -> fail ()
 let tmap3 f (a, b, c) = (f a, f b, f c)
 let tmap4 f (a, b, c, d) = (f a, f b, f c, f d)
 let ( >> ) f g x = g (f x)
 let scale_color_channel x = x /. 256.
+let scale_color_channel = float_of_int >> scale_color_channel
 
 let set_color color =
   match !context with
   | Some ctx ->
-      let r, g, b = tmap3 (float_of_int >> scale_color_channel) color in
-      Cairo.set_source_rgba ctx.ctx r g b 1.
+      let r, g, b = tmap3 scale_color_channel color in
+      Cairo.set_source_rgb ctx.ctx r g b
   | None -> fail ()
 
 (* sets background color *)
 let background color =
   match !context with
-  | Some ctx ->
-      let r, g, b, a = tmap4 (float_of_int >> scale_color_channel) color in
-      Cairo.set_source_rgba ctx.ctx r g b a;
-      Cairo.paint ctx.ctx
+  | Some { ctx; _ }->
+      let r, g, b, alpha = tmap4 scale_color_channel color in
+      Cairo.set_source_rgb ctx r g b;
+      Cairo.paint ctx ~alpha;
+      Cairo.fill ctx
   | None -> fail ()
 
 (** Sets the width of lines for both stroke of shapes and line primitives. 
     Can be any positive integer, with larger numbers producing thicker lines. 
     default is 2 *)
 let set_line_width line_width =
   match !context with
-  | Some ctx -> Cairo.set_line_width ctx.ctx (float_of_int line_width /. 1000.)
+  | Some ctx -> Cairo.set_line_width ctx.ctx (float_of_int line_width)
```

Comment:
Are the lines 1000x wider now?

Reply:
No, because the scaling is 0 to size (ignoring negative offset) instead of 0 to 1. This is how all the Cairo examples work and testing so far points to it being no different.